lib/flash_flow/release/percy_client.rb

```diff
@@ -40,26 +46,33 @@ def find_commit_by_sha(response, sha)
       end
 
       def find_build_by_commit_id(response, commit_id)
-        builds = builds_collection(response)
-        return if builds.nil?
-
-        latest_build = builds
+        builds_collection(response)
           .select { |b| b.dig('relationships', 'commit', 'data', 'id') == commit_id }
           .sort_by { |b| DateTime.parse(b.dig('attributes', 'created-at')) }.last
-
-        latest_build['attributes']
+          &.fetch('attributes', {}) || {}
       end
 
       def builds_collection(response)
-        response['data'].select do |h|
-          h['type'] == 'builds' &&
+        response.fetch('data', [])
+          .select do |h| h['type'] == 'builds' &&
             h.dig('attributes', 'web-url') &&
             h.dig('relationships', 'commit', 'data', 'type') == 'commits'
-        end
+          end
       end
 
       def commits_data(response)
-        response['included'].select { |data| data['type'] == 'commits' }
+        response.fetch('included', {'id' => nil})
+          .select { |data| data['type'] == 'commits' }
+      end
+
+      def has_unapproved_diffs?(build)
+        build['total-comparisons-diff'] > 0 && !build['approved-at'].nil?
```

Comment:
Doesn't this mean that the build is approved? `!build['approved-at'].nil?`.

So you're saying it has unapproved diffs if the build is approved?